hepcrawl/spiders/mit_spider.py

```diff
@@ -0,0 +1,244 @@
+
+    def __init__(self, source_file=None, year=today, *args, **kwargs):
+        """Construct MIT spider"""
+        super(MITSpider, self).__init__(*args, **kwargs)
+        self.source_file = source_file
+        self.year = year
+
+    def start_requests(self):
+        """You can also run the spider on local test files"""
+        if self.source_file:
+            yield Request(self.source_file)
+        elif self.start_urls:
+            html_file = self.get_list_file(self.year)
+            yield Request(html_file)
+
+    def get_list_file(self, year, n=100):
+        """Get data out of the query web page and save it locally."""
+        post_data = {
+            "year": year,  # year, default=current
+            "sort_by": "2",  # sort by date
+            "rpp": n,  # n results per page, default=100
+        }
+        url = self.start_urls[0]
+        req = requests.post(url, data=post_data)
+
+        listing_file = get_temporary_file(prefix="MIT_", suffix=".html")
+
+        with open(listing_file, "w") as outfile:
+            outfile.write(req.text)
+
+        call_uri = u"file://{0}".format(listing_file)
+        return call_uri
+
+    @staticmethod
+    def get_authors(node):
+        """Return authors dictionary """
+        authors_raw = node.xpath(
+            "//td[contains(text(), 'dc.contributor.author')]/following-sibling::td[1]/text()").extract()
+        affiliation = node.xpath(
+            "//td[contains(text(), 'dc.contributor.department')]/following-sibling::td[1]/text()").extract_first()
+
+        authors = []
+        pattern = re.compile(r"(.*)\,\s\d{4}.?")  # Might contain birthdate
+        for author in authors_raw:
+            try:
+                author = pattern.search(author).group(1)
+            except AttributeError:
+                pass
+            surname, given_names = split_fullname(author)
+            authdict = {
+                'surname': surname,
+                'given_names': given_names,
+            }
+            if affiliation:
+                authdict["affiliations"] = [{"value": affiliation}]
+            authors.append(authdict)
+
+        return authors
+
+    def add_fft_file(self, pdf_files, file_access, file_type):
+        """Create a structured dictionary to add to 'files' item."""
+        file_dicts = []
+        for link in pdf_files:
+            file_dict = {
+                "access": file_access,
+                "description": self.name,
+                "url": urljoin(self.domain, link),
+                "type": file_type,
+            }
+            file_dicts.append(file_dict)
+        return file_dicts
+
+    @staticmethod
+    def get_thesis_info(node):
+        """Create thesis info dictionary."""
+        date = node.xpath(
+            "//td[contains(text(), 'dc.date.issued')]/following-sibling::td[1]/text()").extract_first()
+        university = node.xpath(
+            "//td[contains(text(), 'dc.publisher')]/following-sibling::td[1]/text()").extract()
```

Comment:
I believe this should be `extract_first()` as the data model cannot have `thesis.university` be a list.